Implement onboarding review (#551)
returntocorp/semgrep-docs: docs/running-rules.md

## Introduction

> Rules are specific patterns based on which Semgrep reports findings in code. These findings may help you to catch issues of security, performance, correctness, and other bugs. Rules are stored in an open-source [Semgrep Registry](https://github.com/returntocorp/semgrep-rules) that enables you to scan code without need to write anything custom. The list below covers different kinds of Semgrep rules:

Reviewer: "Rules are specific patterns based on which Semgrep reports findings in code"

I think this first sentence is already hard to read and may need a rewrite. 

Maybe something like:
"Semgrep reports findings in code based on rules, which use specific patterns to match code."

Reviewer: Sure, works for me 👍 

Minor grammar error to fix:

> Rules are stored in an open-source [Semgrep Registry](https://github.com/returntocorp/semgrep-rules) that enables you to scan code without need to write anything custom.

Change to:
Rules are stored in an open-source [Semgrep Registry](https://github.com/returntocorp/semgrep-rules) that enables you to scan code without the need to write anything custom.

Author: Changing, thank you!

- Use existing [Semgrep Registry rules](#running-semgrep-registry-rules) or add new rules to Semgrep Registry and run them. See [Contributing to Semgrep rules](https://semgrep.dev/docs/contributing/contributing-to-semgrep-rules-repository/) to find out how to add new rules.
- Local rules:
  - Create and run one-off [ephemeral rules](#ephemeral-rules) in the command line.
  - Create and run [YAML-defined rules in a file](#creating-and-using-yaml-defined-rules-file).
- Run your [local rules simultaneously with Semgrep Registry rules](#running-multiple-rules-simultaneously). 

You can run all rules on your local code or continuously in your Source Code Management (SCM) service (such as GitHub or GitLab) with Semgrep in CI. For more information, see [Semgrep CI overview](semgrep-ci/overview.md).

## Running Semgrep Registry rules locally

Try to run the following example of [Semgrep Registry](https://semgrep.dev/explore) rules:

> 1. Test how Semgrep automatically surveys languages and frameworks and run recommended rules for your source code:

Reviewer: Number (1) and Number (2) are two different procedures. (1) Is a single step procedure, since all they need to run semgrep is to run `semgrep --config auto`

Should be:

A. Test how Semgrep automatically surveys...
B. Explore the Semgrep Registry and choose a rule..
 B.1 On the page of the rule, click **Run Locally**...
 ... up until B.4 Optional

Reviewer: Sure!

    ```sh
    semgrep --config=auto path/to/src
    ```
2. Explore the [Semgrep Registry](https://semgrep.dev/explore) and choose a rule.
3. On the page of the rule, click **Run Locally**.
4. Copy the code for local install, and then add the path to the source code you want to check:
    <pre class="language-bash"><code>semgrep --config="<span className="placeholder">RULESET-ID</span>" <span className="placeholder">PATH/TO/SRC</span></code></pre>
5. Optional: Run registry rules simultaneously with local rules:
   <pre class="language-bash"><code>semgrep --config="<span className="placeholder">RULESET-ID</span>" --config=<span className="placeholder">PATH/TO/YML PATH/TO/SRC</span></code></pre>

:::note
* `--config auto` sends your repository's project URL to [Semgrep Registry](https://semgrep.dev/r) to find rules configured for your repository and as a key for cached rule recommendations.
* When Semgrep Registry is used, [usage metrics](../metrics) are collected by default.
:::

> Add rulesets to Semgrep in CI scans using their **Add to Policy** button on Semgrep Community and Semgrep Team tiers.

Reviewer: Fairly certain this is outdated and superceded by "Add to Rule Board". If this is beyond the scope of this doc we can link to the docs for the rule board (suggestion only).

Reviewer: It depends on how far along we think the user is at their journey. Some users are **prospecting a path** and some users are already decided on what they want to do. When using Semgrep, they need to know what their options are so they can also frame their goal.

Author: I would lead readers more to Semgrep App getting started... this seems like a less heavy solution...

Author: @s-santillan but... we can still include this, I just think that it is a bit of information overload to see all of these links in the docs. :D 

```markdown
:::info
Try Semgrep [Rule Board](semgrep-app/rule-board.md) to determine which rules Semgrep uses and what action Semgrep undertakes when it generates a finding.
:::
```

## Introduction

Rules are specific patterns based on which Semgrep reports findings in code. These findings may help you to catch issues of security, performance, correctness, and other bugs. Rules are stored in an open-source [Semgrep Registry](https://github.com/returntocorp/semgrep-rules) that enables you to scan code without need to write anything custom. Use Semgrep rules in the following ways:

> - Use existing [Semgrep Registry rules](#running-semgrep-registry-rules) or add new rules to Semgrep Registry and run them.

Reviewer 2: also, a link on the "add new rules"?

> - Use existing [Semgrep Registry rules](#running-semgrep-registry-rules) or add new rules to Semgrep Registry and run them.

Author: Yeah I know. :-) Was not sure here, because when we say different kinds of rules, we are only describing them on the basis of how they are set up (in YAML, registry etc.), but not their use (Python rules, C rules, frameworks etc.). Do you have any idea here?

```
 The list below covers different kinds of Semgrep rules:

- Use existing [Semgrep Registry rules](#running-semgrep-registry-rules) or add new rules to Semgrep Registry and run them.
- Local rules:
  - Create and run one-off [ephemeral rules](#ephemeral-rules) in the command line.
  - Create and run [YAML-defined rules in a file](#creating-and-using-yaml-defined-rules-file).
- Run your [local rules simultaneously with Semgrep Registry rules](#running-multiple-rules-simultaneously). 
```

> - Use existing [Semgrep Registry rules](#running-semgrep-registry-rules) or add new rules to Semgrep Registry and run them.

Author: Current state:

```markdown
The list below covers different kinds of Semgrep rules:

- Existing [Semgrep Registry rules](#running-semgrep-registry-rules). See [Contributing to Semgrep rules](https://semgrep.dev/docs/contributing/contributing-to-semgrep-rules-repository/) to find out how to add new rules.
- Local rules:
  - One-off [ephemeral rules](#ephemeral-rules) used in the command line.
  - The [YAML-defined rules](#creating-and-using-yaml-defined-rules-file) in a file.
- Combination of [local rules with Semgrep Registry rules](#running-multiple-rules-simultaneously). 
```

- Local rules:
  - Create and run [ephemeral rules](#ephemeral-rules) in the command line.
  - Create and run [YAML-defined rules](#creating-and-using-yaml-defined-rules-file).

> - Run your [local rules simultaneously with Semgrep Registry rules](#running-multiple-rules-simultaneously). 

Reviewer 2: I will also say, the original "Run Registry Rules" section _sort of_ had a "quick reference" section in its code block. It was confusing since it mixed the comments explaining what you could do with the code itself, however, I wonder if listing the commands below the bullet points in this new section would be helpful.

Author: @efortuna is the list sufficient here? Not sure. :-) This new section (do you mean: `## Running Semgrep Registry rules`?) is not trying to give a reference, but rather really say how to run Semgrep registry rules...

Reviewer 2: yeah we're good!

> You can run all rules on your local code or continuously in your Source Code Management (SCM) service (such as GitHub or GitLab) with Semgrep in CI.

Author: Only a paragraph, not a bullet point. :-) Adding!

```
You can run all rules on your local code or continuously in your Source Code Management (SCM) service (such as GitHub or GitLab) with Semgrep in CI. For more information, see [Semgrep CI overview](semgrep-ci/overview.md).
```

The following diagram displays the current number of Semgrep rules:
<div className="lang-container" style={{marginBottom: '20px'}}>

>   <iframe width="900" height="400" frameBorder="0" src="https://dashboard.semgrep.dev/metric/semgrep-rules.num/graph"></iframe>

Author: True true, it is not very clean with that black part... and might give too much information when folks want to try out rules. Maybe I can move his diagram to some appendix or: The interesting facts about rules section at the bottom of this document. Not sure how to create the counter. But it would be great here. @efortuna

>   <iframe width="900" height="400" frameBorder="0" src="https://dashboard.semgrep.dev/metric/semgrep-rules.num/graph"></iframe>

Author: Moving at the bottom of the article to a new section.

</div>

> ## Running Semgrep Registry rules

Reviewer 2: I didn't see anywhere talking about running rules non-locally. If we don't, depending on how things get structured, you might want to change this title to "Running Semgrep Registry rules locally", and then you can take out that first sentence below. Also if we want to talk about running not locally, can we link to the CI page if people want that?

Author: Ok, changes done as you suggested, makes sense! I have one uncertainty though. To run Semgrep non-locally, we need to integrate Semgrep App with CI, right? Not sure if CI overview is so much helpful here. Maybe Semgrep App getting started might be a nicer link here? :-) @efortuna

Reviewer 2: hmm I still think the CI overview page https://semgrep.dev/docs/semgrep-ci/overview says you need the app and it makes the most sense to link to the CI page. But I defer to @s-santillan for the final opinion.

Author: @efortuna adding this as a subsection, so under the step 4. in the procedure described in this section, there will be:

```markdown
### Next steps

To use Semgrep Registry continuously in your CI/CD pipeline, see the documentation for [Semgrep in CI](https://semgrep.dev/docs/semgrep-ci/overview).
```

> This section explains how to run [Semgrep Registry](https://semgrep.dev/explore) rules locally. See the following examples of how you may test Semgrep rules:

Reviewer 2: "The following examples" is a bit confusing since we start below with a list of directions. Which examples are you trying to reference here?

Author: I see, it was wrong really. :D 
Changing to: `Try to run the following example of [Semgrep Registry](https://semgrep.dev/explore) rules:`

1. Test how Semgrep automatically surveys languages and frameworks and run recommended rules for your source code:
    ```sh
    semgrep --config=auto path/to/src
    ```
2. Explore the [Semgrep Registry](https://semgrep.dev/explore) and choose a rule.
3. On the page of the rule, click **Run Locally**.
4. Copy the code for local install, and then add the path to the source code you want to check:
    <pre class="language-bash"><code>semgrep --config="<span className="placeholder">RULESET-ID</span>" <span className="placeholder">PATH/TO/SRC</span></code></pre>
5. Optional: Run registry rules simultaneously with local rules:
   <pre class="language-bash"><code>semgrep --config="<span className="placeholder">RULESET-ID</span>" --config=<span className="placeholder">PATH/TO/YML PATH/TO/SRC</span></code></pre>

:::note
* `--config auto` sends your repository's project URL to [Semgrep Registry](https://semgrep.dev/r) to find rules configured for your repository and as a key for cached rule recommendations.
* When Semgrep Registry is used, [usage metrics](../metrics) are collected by default.
:::

> Add rulesets to Semgrep in CI scans using their "Add to Policy" button on Semgrep Community and Semgrep Team tiers.

Reviewer 2: is there a relevant docs page to link to here?

## Introduction

Semgrep reports findings in code based on rules. These rules use specific patterns to match code. When Semgrep matches a code, it is reported as a finding. Findings may help you to catch issues of security, performance, and correctness. Rules are stored in an open-source [Semgrep Registry](https://github.com/returntocorp/semgrep-rules) that enables you to scan code without need to write anything custom. The list below covers different kinds of Semgrep rules:

> - Existing [Semgrep Registry rules](#running-semgrep-registry-rules). See [Contributing to Semgrep rules](https://semgrep.dev/docs/contributing/contributing-to-semgrep-rules-repository/) to find out how to add new rules.

Author: @efortuna Included your suggestion! I will add your text about a large number of rules below close to the diagram. :-)

- Local rules:
  - One-off [ephemeral rules](#ephemeral-rules) used in the command line.

>   - The [YAML-defined rules](#creating-and-using-yaml-defined-rules-file) in a file.

Reviewer 3: however the output of semgrep also gets a bit simplified so it looks closer to `grep` in this mode; mentions of rules and rule IDs are not printed in this mode

>   - The [YAML-defined rules](#creating-and-using-yaml-defined-rules-file) in a file.

Author: @underyx thank you! :-)

>   - The [YAML-defined rules](#creating-and-using-yaml-defined-rules-file) in a file.

Author: To apply @efortuna suggestion using its breadth - I will shorten this link to: `[YAML-defined rules](#yaml-defined-rules).`

>   - The [YAML-defined rules](#creating-and-using-yaml-defined-rules-file) in a file.

Reviewer 2: got it. do what @underyx says then!

- Combination of [local rules with Semgrep Registry rules](#running-multiple-rules-simultaneously). 

You can run all rules on your code locally or continuously in your Source Code Management (SCM) service (such as GitHub or GitLab) with Semgrep in CI. For more information, see [Semgrep CI overview](semgrep-ci/overview.md).

## Running Semgrep Registry rules locally

> Test how Semgrep automatically surveys languages and frameworks and run recommended rules for your source code (substitute the `PATH/TO/SRC` with path to your source code):

Reviewer 2: ```suggestion
You can test how Semgrep automatically surveys languages and frameworks and run recommended rules for your source code with the following command (substitute the `PATH/TO/SRC` with path to your source code):
```

Reviewer 2: Giving this suggestion because as written, it sounds like a command "Test!".

Reviewer 2: Alsooo, I think the (substitute the `PATH/TO/SRC` with path to your source code) is probably understandable? But @s-santillan can give her opinion on this.

Reviewer 2: I think the question I have is, is this part exactly procedural? the part below it is. But this is leading you into it. You can also phrase it as "To test how semgrep....run:"

    ```sh
    semgrep --config=auto path/to/src
    ```
2. Explore the [Semgrep Registry](https://semgrep.dev/explore) and choose a rule.
3. On the page of the rule, click **Run Locally**.
4. Copy the code for local install, and then add the path to the source code you want to check:
    <pre class="language-bash"><code>semgrep --config="<span className="placeholder">RULESET-ID</span>" <span className="placeholder">PATH/TO/SRC</span></code></pre>
5. Optional: Run registry rules simultaneously with local rules:
   <pre class="language-bash"><code>semgrep --config="<span className="placeholder">RULESET-ID</span>" --config=<span className="placeholder">PATH/TO/YML PATH/TO/SRC</span></code></pre>
Test how Semgrep automatically surveys languages and frameworks and run recommended rules for your source code (substitute the `PATH/TO/SRC` with path to your source code):
<pre class="language-bash"><code>semgrep --config=auto <span className="placeholder">PATH/TO/SRC</span></code></pre>

:::note
:::info
* `--config auto` sends your repository's project URL to [Semgrep Registry](https://semgrep.dev/r) to find rules configured for your repository and as a key for cached rule recommendations.

> * When Semgrep Registry is used, [usage metrics](../metrics) are collected by default.

Reviewer 2: @brendongo suggests on slack that we are still collecting these metrics. Are the release notes wrong or referring to something else?

Reviewer: Yes we are still collecting these metrics. Keep the line in please!

Author: Returned yesterday already. GitHub keeps it here because of the comment, it is not in source actually.

>    <pre class="language-bash"><code>semgrep --config="<span className="placeholder">RULESET-ID</span>" --config=<span className="placeholder">PATH/TO/YML PATH/TO/SRC</span></code></pre>

Reviewer 2: maybe PATH/TO/REGISTRY/SRC and PATH/TO/LOCAL/YML just to clarify? I guess it doesn't matter the order, but it took me a minute since I was like "but they're both yaml..."

Author: @efortuna What about:

```bash
semgrep --config="RULESET-ID" --config=PATH/TO/RULE.YAML PATH/TO/SRC
```

- Existing [Semgrep Registry rules](#running-semgrep-registry-rules) or [rules contributed to the Semgrep Registry by you](https://semgrep.dev/docs/contributing/contributing-to-semgrep-rules-repository/) to find out how to add new rules.
- [Local rules](#creating-and-using-local-rules):
  - One-off [ephemeral rules](#ephemeral-rules) passed into the command line.
  - [YAML-defined rules](#yaml-defined-rules).
- A combination of [local rules and Semgrep Registry rules](#running-multiple-rules-simultaneously). 

You can run all rules on your code completely locally or continuously in your Source Code Management (SCM) service (such as GitHub or GitLab) with Semgrep in CI. For more information, see [Semgrep CI overview](semgrep-ci/overview.md).

## Running Semgrep Registry rules locally

> You can test how Semgrep automatically surveys languages and frameworks and run recommended rules for your source code with the following command (substitute the `PATH/TO/SRC` with path to your source code):

Reviewer 4: This is not accurate. As of now we do not run surveys to recommend rules.

Author: @brendongo what is the logic to determine the language or framework now? @brendongo What do we do instead of surveys?

Reviewer 4: We send the same rules no matter what

- Existing [Semgrep Registry rules](#running-semgrep-registry-rules) or [rules contributed to the Semgrep Registry by you](https://semgrep.dev/docs/contributing/contributing-to-semgrep-rules-repository/) to find out how to add new rules.
- [Local rules](#creating-and-using-local-rules):
  - One-off [ephemeral rules](#ephemeral-rules) passed into the command line.
  - [YAML-defined rules](#yaml-defined-rules).
- A combination of [local rules and Semgrep Registry rules](#running-multiple-rules-simultaneously). 

You can run all rules on your code completely locally or continuously in your Source Code Management (SCM) service (such as GitHub or GitLab) with Semgrep in CI. For more information, see [Semgrep CI overview](semgrep-ci/overview.md).

## Running Semgrep Registry rules locally

You can test Semgrep Registry rules with the following command:
<pre class="language-bash"><code>semgrep --config=auto <span className="placeholder">PATH/TO/SRC</span></code></pre>
Substitute the `PATH/TO/SRC` with the path to your source code.

:::info

> * `--config auto` sends your repository's project URL to [Semgrep Registry](https://semgrep.dev/r) to find rules configured for your repository and as a key for cached rule recommendations.

Reviewer 4: We can remove this

Author: Removed!

## Introduction

> Semgrep reports findings in code based on rules. These rules use specific patterns to match code. When Semgrep matches a portion of code, it is reported as a finding. Findings may help you to catch issues of security, performance, correctness, and best practices. Rules are stored in an open-source [Semgrep Registry](https://github.com/returntocorp/semgrep-rules) that enables you to scan code without need to write anything custom. The list below covers different kinds of Semgrep rules:

Reviewer 2: The way you had this previously started with what rules are in the first sentence, and now we wait until the second sentence. Did I miss a comment that suggested this part should be rearranged? I think it's clearer to define what rules are up-front

Author: @efortuna ok! Please, feel free to suggest an edit here.

Author: @efortuna So Sara suggested the original [thing here](https://github.com/returntocorp/semgrep-docs/pull/551#discussion_r901233736) - so the current state of this paragraph is:

```markdown
Semgrep reports findings in code based on rules, which use specific patterns to match code. When Semgrep matches a portion of code, it is reported as a finding. Findings may help you to catch issues of security, performance, correctness, and best practices. Rules are stored in an open-source [Semgrep Registry](https://github.com/returntocorp/semgrep-rules) that enables you to scan code without the need to write anything custom. The list below covers different kinds of Semgrep rules:
```

Reviewer 2: just to follow up on this comment, I made a suggestion in the latest version.

## Running Semgrep Registry rules locally

You can test how Semgrep automatically surveys languages and frameworks and run recommended rules for your source code with the following command (substitute the `PATH/TO/SRC` with path to your source code):
<pre class="language-bash"><code>semgrep --config=auto <span className="placeholder">PATH/TO/SRC</span></code></pre>

:::info
* `--config auto` sends your repository's project URL to [Semgrep Registry](https://semgrep.dev/r) to find rules configured for your repository and as a key for cached rule recommendations.
* When Semgrep Registry is used, [usage metrics](metrics.md) are collected by default.
:::

Explore the Semgrep Registry by following these steps:

1. See the [Semgrep Registry](https://semgrep.dev/explore) and choose a rule.
2. On the page of the rule, click **Run Locally**.
3. Copy the code for local install, and then add the path to the source code you want to check in your terminal:
    <pre class="language-bash"><code>semgrep --config="<span className="placeholder">RULESET-ID</span>" <span className="placeholder">PATH/TO/SRC</span></code></pre>
4. Optional: Run registry rules simultaneously with local rules:

>    <pre class="language-bash"><code>semgrep --config="<span className="placeholder">RULESET-ID</span>" --config=<span className="placeholder">PATH/TO/MYRULE.YAML PATH/TO/SRC</span></code></pre>

Reviewer 2: I'll suggest PATH/TO/REGISTRY/SRC once more and then stop suggesting it if you prefer the other way

Reviewer 2: oop, sorry I see you had that previous question "unresolved" I must have had my github view wrong. Sorry about that!

Author: I mean... a path to SRC is a path to the source code you want to have checked... the path to myrule.yaml is a path to a local YAML file with rules... so these are two different things. But it might be confusing really! So what about:

`PATH/TO/SEMGREP_RULES.YAML PATH/TO/SRC`

Or maybe I am missing something and you mean the `PATH/TO/REGISTRY/SRC` for the `RULESET-ID`? :-)

Reviewer 2: omg, no I misunderstood. thanks for the clarification! I thought for some reason the SRC was you downloading the rules from the registry. ignore my comments!

Author: No worries, maybe more users will have problem with this... Would be great to one day try this with somebody who have not tried Semgrep before. @efortuna

## Introduction

> Semgrep reports findings in code based on rules, which use specific patterns to match code. When Semgrep matches a portion of code, it is reported as a finding. Findings may help you to catch issues of security, performance, correctness, and best practices. Rules are stored in an open-source [Semgrep Registry](https://github.com/returntocorp/semgrep-rules) that enables you to scan code without the need to write anything custom. The list below covers different kinds of Semgrep rules:

Reviewer 2: Suggestion: 
Rules are used to detect specific patterns in a codebase. Semgrep reports all the matched locations as findings. Such findings cn help you catch issues of security...[the rest is unchanged]

Author: I understand your point @efortuna. I am still thinking that we are not paying our rules justice... they are recipes in the end, right? Recipes to find stuff in code (since it can be only one file and not necessarily the whole codebase, I would rather say "code", as it involves both single files and whole codebases in its meaning). What about:

Suggestion based on your suggestion:
Rules are instructions based on which Semgrep detects patterns in code. When Semgrep reports a code using mentioned rules, the detected code is called a finding. The process of scanning and detecting a piece of code is sometimes called *matching*, as Semgrep matched the code using rules to report a finding. Such findings can help you catch issues of security, performance, correctness, and best practices. Rules also have a form of patterns that you can define, or you can rely on rules created by the community or r2c. Rules are stored in an open-source [Semgrep Registry](https://github.com/returntocorp/semgrep-rules) that enables you to scan code without the need to write anything custom. The list below covers different kinds of Semgrep rules:

Reviewer: Original
"Rules also have a form of patterns that you can define, or you can rely on rules..."

Suggested
 -> You can define custom rules through Semgrep's rule syntax or you can rely on rules created by the community or r2c.

Author: Implemented!

```

:::info

> Both local yaml files and ephemeral rules are called *local rules*.

Reviewer 2: ```suggestion
Both local rule.yaml files and ephemeral rules are called *local rules*.
```

```

:::info
Both local `rule.yaml` files and ephemeral rules are called *local rules*.
:::

### YAML-defined rules

> Create a YAML file where you define your own rules and run them with Semgrep. To make your own local rule file, follow these steps:

Reviewer 2: Here's another one where starting with a verb sounds like a command. Did you mean this to read instead like "You can create a YAML..." ? Though the first sentence feels relatively self-explanatory.

Author: Oh yeah, you are right @efortuna. These imperatives should be reserved for list items with steps... Changing it to you can. :-)

Reviewer 2: I'd argue this part isn't  "procedural" (but tell me if I'm wrong). 
If you don't like the "you can " starting with, how about:

```suggestion
To create a local YAML file where you define your own rules and run them with Semgrepe, follow these steps:
```

(it's a little long, and we know it's with Semgrep, but ¯\\\_(ツ)_/¯

Author: Great! Including it. :-)

      pattern: $SOMEVAR is "..."
      severity: ERROR
    ```
3. Run the following command to run local YAML rule files:
    ```sh
    semgrep --config path/to/rule.yaml
    ```

For more information, see [Getting started](../writing-rules/overview/).

## Running multiple rules simultaneously

To run multiple rules simultaneously, use `--config` before every YAML URL, or Semgrep registry entry name. This option let's you include your local rules as well as Semgrep Registry rules. See the following code example (substitute the colored values as necessary):

<pre class="language-bash"><code>semgrep --config <span className="placeholder">p/python</span> --config <span className="placeholder">MYRULES/MYRULE.YAML</span></code></pre>

## Findings

* See [Managing findings](../managing-findings/) for information on Semgrep findings.
* See [Ignoring findings](../ignoring-files-folders-code/) for details on suppressing rule output.

## Next steps

> Contribute to the registry by writing your own rules and adding them to the <a href="https://github.com/returntocorp/semgrep-rules" target="_blank">Semgrep rules repository</a>.

Reviewer 2: nice!

Reviewer 2: didn't you have a docs page on how to contribute to the registry?

Author: Changing to:

```
Contribute to the registry by writing your own rules and adding them to the <a href="https://github.com/returntocorp/semgrep-rules" target="_blank">Semgrep rules repository</a>. Find out how to contribute to Semgrep rules repository by reading [Contributing to Semgrep rules](contributing/contributing-rules.md) guide.
```
